src/debugpy/server/tracing/__init__.py

```diff
@@ -375,42 +373,61 @@ def invalidate(self, thread: Thread):
             frame.python_frame = None
 
 
-@dataclass
 class Step:
-    step: Literal["in", "out", "over"]
-    origin: FrameType = None
-    origin_line: int = None
+    def __init__(
+        self,
+        step: Literal["in", "out", "over", "goto"],
```

Comment:
"goto" is not needed anymore

Reply:
Also I think this can be made simpler by making it a class variable and just setting it in derived classes, i.e.:
```py
class Step:
   kind: str
   ...

class StepIn(Step):
   kind = "in"
   ...
```